### PR TITLE
Submit form json to submit_claim endpoint

### PR DESCRIPTION
**Pull request by cesaresantiago**
## Description
This code parses the information on the form, transforms it before form submission and sends the JSON request expected by the submit_claim endpoint.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/my-education-benefits/config/form.js [hunk 1]
```diff
@@ -36,9 +36,11 @@ import PhoneReviewField from '../components/PhoneReviewField';
 import DateReviewField from '../components/DateReviewField';
 import EmailReviewField from '../components/EmailReviewField';
 
+import environment from 'platform/utilities/environment';
+
 import {
-  activeDutyLabel,
-  selectedReserveLabel,
+  chapter30Label,
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:42]: I changed this labels to have meaningful names. It was hard to track that they were related to chapter30 and chapter 1606 relinquisments.

File: src/applications/my-education-benefits/config/form.js [hunk 2]
```diff
@@ -72,7 +76,7 @@ const formFields = {
   dateOfBirth: 'dateOfBirth',
   ssn: 'ssn',
   toursOfDuty: 'toursOfDuty',
-  toursOfDutyCorrect: 'toursOfDutyCorrect',
+  serviceHistoryIncorrect: 'serviceHistoryIncorrect',
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:79]: The flag name gave the impression that if it was true, the service history was correct. In fact, it was the opposite, so changed name to reflect that.

File: src/applications/my-education-benefits/config/form.js [hunk 3]
```diff
@@ -85,7 +89,7 @@ const formFields = {
   phoneNumber: 'phoneNumber',
   mobilePhoneNumber: 'mobilePhoneNumber',
   viewBenefitSelection: 'view:benefitSelection',
-  benefitSelection: 'benefitSelection',
+  benefitRelinquished: 'benefitRelinquished',
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:92]: The name gave impression that it listed benefits selected to apply for, when in fact, it referred to relinquished benefits. Changed name to match.

File: src/applications/my-education-benefits/config/form.js [hunk 4]
```diff
@@ -214,7 +218,7 @@ function phoneSchema() {
 }
 
 function additionalConsiderationsQuestionTitleText(benefitSelection, order) {
-  const isUnsure = !benefitSelection || benefitSelection === 'UNSURE';
+  const isUnsure = !benefitSelection || benefitSelection === 'CannotRelinquish';
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:221]: The expected value by the automation service if we select "Not sure" is "CannotRelinquish". Changed to match.

File: src/applications/my-education-benefits/config/form.js [hunk 5]
```diff
@@ -305,8 +309,8 @@ function AdditionalConsiderationTemplate(page, formField) {
 }
 
 function givingUpBenefitSelected(formData) {
-  return ['ACTIVE_DUTY', 'SELECTED_RESERVE'].includes(
-    formData[formFields.viewBenefitSelection][formFields.benefitSelection],
+  return ['Chapter30', 'Chapter1606'].includes(
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:312]: Changed names to reflect that these labels referred to the actual chapters to be relinquished. It was hard to figure out with current naming.

File: src/applications/my-education-benefits/config/form.js [hunk 6]
```diff
@@ -337,11 +341,16 @@ function renderContactMethodFollowUp(formData, cond) {
   return res;
 }
 
+function transform(metaData, form) {
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:344]: This transform function is called presubmission and reads the form data and turns it into expected JSON.

File: src/applications/my-education-benefits/config/form.js [hunk 7]
```diff
@@ -337,11 +341,16 @@ function renderContactMethodFollowUp(formData, cond) {
...
+  const submission = createSubmissionForm(form.data);
+  return JSON.stringify(submission);
+}
+
 const formConfig = {
   rootUrl: manifest.rootUrl,
   urlPrefix: '/',
-  submit: () =>
-    Promise.resolve({ attributes: { confirmationNumber: '123123123' } }),
+  submitUrl: `${environment.API_URL}/meb_api/v0/submit_claim`,
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:352]: Removed mock submit and changed it to actual endpoint.

File: src/applications/my-education-benefits/config/form.js [hunk 8]
```diff
@@ -337,11 +341,16 @@ function renderContactMethodFollowUp(formData, cond) {
...
+  transformForSubmit: transform,
   trackingPrefix: 'my-education-benefits-',
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/config/form.js:354]: This config options tells the form system to run the transform option pre-submission.

File: src/applications/my-education-benefits/tests/fixtures/data/form-submission-test-data.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
+export const submissionForm = {
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/tests/fixtures/data/form-submission-test-data.js:1]: This is test form data for unit tests based on what I saw in memory with the debugger. This data will be parsed by transform and changed to meaningful JSON for the service.

File: src/applications/my-education-benefits/tests/form-submit-transform.unit.spec.js [hunk 10]
```diff
@@ -0,0 +1,313 @@
+import { expect } from 'chai';
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/tests/form-submit-transform.unit.spec.js:1]: Unit tests for each method in the transform utility functions to parse the form data and turn it into JSON.

File: src/applications/my-education-benefits/utils/form-submit-transform.js [hunk 11]
```diff
@@ -0,0 +1,393 @@
+// this constant maps the values on address.js in vets.json schema from VA.gov values to LTS values
+// the lts values were found on the LTS database and LTS validates them, so we need to send correct value from here
+const countries = [
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/utils/form-submit-transform.js:3]: This list is used to map VA.gov three letter code for each country to the equivalent codes on LTS. To figure out the codes I went into the LTS country code Database and found each two letter code for each country.

During transformation, we'll map the three letter code value that comes from va.gov country pull down menu, and map it to LTS code and send on JSON.

File: src/applications/my-education-benefits/utils/form-submit-transform.js [hunk 12]
```diff
@@ -0,0 +1,393 @@
...
+];
+
+export function getLTSCountryCode(schemaCountryValue) {
+  const country = countries.find(countryInfo => {
+    return countryInfo.schemaValue === schemaCountryValue;
+  });
+  return country?.ltsValue ? country.ltsValue : 'ZZ'; // ZZ is LTS code for unknown
+}
+
+export function getAddressType(mailingAddress) {
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/utils/form-submit-transform.js:257]: Transforming to expected LTS values for address types.

File: src/applications/my-education-benefits/utils/form-submit-transform.js [hunk 13]
```diff
@@ -0,0 +1,393 @@
...
+  { schemaValue: 'VGB', ltsValue: 'VI', label: 'British Virgin Islands' },
+  { schemaValue: 'WLF', ltsValue: 'WF', label: 'Wallis and Futuna' },
+  { schemaValue: 'ESH', ltsValue: 'WI', label: 'Western Sahara' },
+  { schemaValue: 'YEM', ltsValue: 'YM', label: 'Yemen' },
+  { schemaValue: 'ZMB', ltsValue: 'ZA', label: 'Zambia' },
+  { schemaValue: 'ZWE', ltsValue: 'ZI', label: 'Zimbabwe' },
+];
+
+export function getLTSCountryCode(schemaCountryValue) {
+  const country = countries.find(countryInfo => {
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/utils/form-submit-transform.js:251]: Using constants above, find correct LTS code for country selected.

File: src/applications/my-education-benefits/utils/form-submit-transform.js [hunk 14]
```diff
@@ -0,0 +1,393 @@
+// this constant maps the values on address.js in vets.json schema from VA.gov values to LTS values
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/utils/form-submit-transform.js:1]: In general, these transform methods will generate the expected JSON request that the claim submission DGIB service expects.

File: src/applications/my-education-benefits/utils/form-submit-transform.js [hunk 15]
```diff
@@ -0,0 +1,393 @@
...
+    }
+  } else {
+    return {
+      claimantComment: {},
+      disagreeWithServicePeriod: false,
+    };
+  }
+}
+
+export function createSubmissionForm(submissionForm) {
```
Comment (cesaresantiago) [inline src/applications/my-education-benefits/utils/form-submit-transform.js:386]: This is the method that will be called on form.js to create the JSON request.

**Review by FernandoBoza — APPROVED** [review]

**Pull request merged**